homeassistant/components/sonos/media_player.py

```diff
@@ -938,8 +938,23 @@ def play_media(self, media_type, media_id, **kwargs):
         """
         Send the play_media command to the media player.
 
+        If media_type is "playlist", media_id should be a Sonos
+        Playlist name.  Otherwise, media_id should be a URI.
+
         If ATTR_MEDIA_ENQUEUE is True, add `media_id` to the queue.
         """
+        if media_type == "playlist":
+            try:
+                playlists = self.soco.get_sonos_playlists()
+                playlist = next(p for p in playlists if p.title == media_id)
+                self.soco.clear_queue()
+                self.soco.add_to_queue(playlist)
+                self.soco.play_from_queue(0)
+                return
+            except StopIteration:
+                _LOGGER.error('Could not find a Sonos playlist named "%s".', media_id)
+                # Fall through to the previous behavior, in case someone is relying on
+                # passing a URI with a media_type of "playlist"
```

Comment:
Let's not dig that hole any deeper :-). Rather, let's enforce "music" or "playlist" media types going forward and mention that as a breaking change instead.